assets/js/base/components/cart-checkout/totals/discount/index.tsx

```diff
@@ -4,18 +4,37 @@
 import { __, sprintf } from '@wordpress/i18n';
 import LoadingMask from '@woocommerce/base-components/loading-mask';
 import { RemovableChip } from '@woocommerce/base-components/chip';
-import PropTypes from 'prop-types';
 import {
 	__experimentalApplyCheckoutFilter,
 	TotalsItem,
 } from '@woocommerce/blocks-checkout';
 import { getSetting } from '@woocommerce/settings';
+import {
+	CartResponseCouponItemWithLabel,
+	CartTotalsItem,
+	Currency,
+} from '@woocommerce/types';
+import { LooselyMustHave } from '@woocommerce/type-defs/utils';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 
+export interface TotalsDiscountProps {
+	cartCoupons: LooselyMustHave<
+		CartResponseCouponItemWithLabel,
+		'code' | 'label' | 'totals'
```

Comment:
`label` is used, e.g., on L 93 and 94. It's basically using to create the text for the chip :)

As for `totals` it is not technically required by the component code itself, but I noticed that's the way it is used in practice, as you need to calculate the total discount by summing up each coupon. An example of it can be seen in the story I wrote.

Reply:
Aha right you are, my bad I missed that